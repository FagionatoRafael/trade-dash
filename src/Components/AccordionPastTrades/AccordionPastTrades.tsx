import Accordion from '@mui/material/Accordion';
import AccordionSummary from '@mui/material/AccordionSummary';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import React from 'react';
import Typography from '@mui/material/Typography';
import AccordionDetails from '@mui/material/AccordionDetails';
import TablePastTrades from '../../Components/tablePastTrades/tablePastTrades';

interface pastTradesParams {
    FromTo: string
      valueUSD: number,
    valueGBP:	number
  }

interface TablePsatTradeProps {
    rowsPast: Array<pastTradesParams>
}

const AccordionPastTrades: React.FC<TablePsatTradeProps> = (props) => {

  return (
  <>
    <Accordion>
      <AccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls="panel1a-content"
        id="panel1a-header"
      >
          <Typography>Past Trades</Typography>
      </AccordionSummary>
      <AccordionDetails>
        <Typography>
            <TablePastTrades rowsPast={props.rowsPast}/>
        </Typography>
      </AccordionDetails>   
    </Accordion>  
  </>
  );
}

export default AccordionPastTrades;