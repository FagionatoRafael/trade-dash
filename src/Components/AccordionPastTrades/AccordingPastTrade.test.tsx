import React from 'react';
import {cleanup, render} from '@testing-library/react';
import AccordionPastTrades from './AccordionPastTrades';
import TablePastTrades from '../tablePastTrades/tablePastTrades'

afterEach(cleanup);

describe('AccordingPastTrade component', () => {
  test('Should render component', () => {
    const renderComponent = render(<AccordionPastTrades rowsPast={[]} />);

    expect(renderComponent).toMatchSnapshot();
  })

  test('should render component TablePastTrades', () => {
    const renderComponent = render(<TablePastTrades rowsPast={[]} />)

    expect(renderComponent).not.toBeUndefined()
  })
  
})