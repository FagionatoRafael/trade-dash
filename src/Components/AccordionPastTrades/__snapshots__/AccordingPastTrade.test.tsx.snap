// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordingPastTrade component Should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="panel1a-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Past Trades
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                class="MuiAccordion-region"
                id="panel1a-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <div
                      class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
                    >
                      <table
                        aria-label="a dense table"
                        class="MuiTable-root css-quj9j6-MuiTable-root"
                      >
                        <thead
                          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                        >
                          <tr
                            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                          >
                            <th
                              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                              scope="col"
                            >
                              Trade
                            </th>
                            <th
                              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                              scope="col"
                            >
                              Value DOLLAR
                            </th>
                            <th
                              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                              scope="col"
                            >
                              value POUND
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                        />
                      </table>
                    </div>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-controls="panel1a-content"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
        id="panel1a-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Past Trades
          </p>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              class="MuiAccordion-region"
              id="panel1a-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <div
                    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
                  >
                    <table
                      aria-label="a dense table"
                      class="MuiTable-root css-quj9j6-MuiTable-root"
                    >
                      <thead
                        class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                      >
                        <tr
                          class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                        >
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
                            scope="col"
                          >
                            Trade
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                            scope="col"
                          >
                            Value DOLLAR
                          </th>
                          <th
                            class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
                            scope="col"
                          >
                            value POUND
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                      />
                    </table>
                  </div>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
